File: 2300 successful-pairs-of-spells-and-potions/solution.go
func successfulPairs(spells []int, potions []int, success int64) []int {
    sort.Ints(potions)
    m := len(potions)
    ans := make([]int, len(spells))

    for i, spell := range spells {
        l, r := 0, m
        for l < r {
            mid := l + (r-l)/2
            v := int64(potions[mid]) * int64(spell)
            if v < success {
                l = mid + 1
            } else {
                r = mid
            }
        }
        ans[i] = m - l
    }
    return ans
}

File: 2300 successful-pairs-of-spells-and-potions/solution.cs
public class Solution {
    public int[] SuccessfulPairs(int[] spells, int[] potions, long t) {
        Array.Sort(potions);
        int n = spells.Length;
        int m = potions.Length;
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            int l=0,r=m;
            while (l < r) {
                int mi = l + (r - l) / 2;
                long v = (long) potions[mi] * spells[i];
                if (v < t)
                    l = mi + 1;
                else
                    r = mi;
            }
            ans[i] = m - l;
        }
        return ans;
    }
}
